mono/utils/mono-time.c

```diff
@@ -161,6 +178,31 @@ mono_100ns_ticks (void)
 }
 
 /*
+ * Returns user + system time for a thread in 100ns units.
+ */
+gint64
+mono_thread_cpu_time (void)
+{
+#if defined(__linux__)
```

Comment:
We avoid using compiler supported defines as they break in cross compilation scenarios, use the ones we produce.
Use PLATFORM_LINUX here.